### PR TITLE
Allow setting group conditions with different trigger modes

### PR DESCRIPTION
**Pull request by dkorn**
This change makes it possible to set group conditions with `FIRING` and `AUTORESOLVE` trigger_mode.

### CODE REVIEW

**Comment by dkorn** [comment]
@cben @simon3z @enoodle @simon3z please review

cc @zgalor

File: create_group_trigger.yml [hunk 1]
```diff
@@ -13,13 +13,18 @@
...
       state: 'present'
       group_conditions:
-        trigger_mode: 'FIRING'
-        conditions:
-          - name: 'Test Condition'
-            type: 'THRESHOLD'
-            data_id: 'test_condition'
-            operator: 'GT'
-            threshold: 0.8
+        - name: 'Test Condition 01'
```
Comment (cben) [inline create_group_trigger.yml]: Excellent, this allowed you to drop the nesting :drum: 
Consider calling it just `conditions:` now.

Comment (dkorn) [reply create_group_trigger.yml]: Done, thanks :)

File: library/hawkular_alerts_group_trigger.py [hunk 2]
```diff
@@ -210,15 +208,19 @@ def required_updates(self, trigger, name, severity, enabled):
...
+        for c in conditions:
             name = c.pop("name")
             condition = hawkular.alerts.Condition(c)
-            condition.trigger_mode = conditions["trigger_mode"]
             condition.context = {'name': name}
-            gc.addCondition(condition)
+            if condition.trigger_mode == "FIRING":
+                firing_gc.addCondition(condition)
+            elif condition.trigger_mode == "AUTORESOLVE":
+                autoresolve_gc.addCondition(condition)
```
Comment (cben) [inline library/hawkular_alerts_group_trigger.py:220]: else: report error

File: library/hawkular_alerts_group_trigger.py [hunk 3]
```diff
@@ -210,15 +208,19 @@ def required_updates(self, trigger, name, severity, enabled):
...
         try:
-            self.client.create_group_conditions(group_id, conditions["trigger_mode"], gc)
+            self.client.create_group_conditions(group_id, "FIRING", firing_gc)
+            self.client.create_group_conditions(group_id, "AUTORESOLVE", autoresolve_gc)
```
Comment (cben) [inline library/hawkular_alerts_group_trigger.py]: Could check each separately whether update required.
Not worth it if complicates code, your call.

File: library/hawkular_alerts_group_trigger.py [hunk 4]
```diff
@@ -237,7 +239,7 @@ def update_group_trigger(self, trigger, updates, conditions):
         except Exception as e:
             self.module.fail_json(msg="Failed to update group trigger. Error: {error}".format(error=e))
         if conditions:
```
Comment (cben) [inline library/hawkular_alerts_group_trigger.py]: Hmm.  I think this should be `if conditions is not None` as below in create_or_update_group_trigger().
If it's explicitly given as empty list, we do want to set_group_trigger_conditions().

**Comment by cben** [comment]
You have duplicate logic to call set_group_trigger_conditions():
1. from update_group_trigger()
2. from create_or_update_group_trigger() in case it's not calling update_group_trigger().
I think it should be possible to drop (1) and "always" do (2).
[well not always but if not None and condition_update_required()...]

File: library/hawkular_alerts_group_trigger.py (hunk 2 above)
Comment (dkorn) [reply library/hawkular_alerts_group_trigger.py:220]: done, thanks

File: library/hawkular_alerts_group_trigger.py (hunk 3 above)
Comment (dkorn) [reply library/hawkular_alerts_group_trigger.py]: you're right, but it does complicate.. 
maybe you can mention it as an issue  for future refactoring?

File: library/hawkular_alerts_group_trigger.py (hunk 4 above)
Comment (dkorn) [reply library/hawkular_alerts_group_trigger.py]: right, thanks!

**Comment by dkorn** [comment]
> You have duplicate logic to call set_group_trigger_conditions():
> 
> from update_group_trigger()
> from create_or_update_group_trigger() in case it's not calling update_group_trigger().
> I think it should be possible to drop (1) and "always" do (2).
> [well not always but if not None and condition_update_required()...]

Thing is, I also have it in `create_group_trigger()`. do you think it'll be better to drop that one as well?
I want to merge this PR, the dampening one is dependent on it.

**Pull request merged**